src/interface/src/app/treatments/treatment-layout/treatment-layout.component.ts

```diff
@@ -73,58 +76,6 @@ export class TreatmentLayoutComponent {
     map(([activeArea, showTreatmentLayer]) => !activeArea && showTreatmentLayer)
   );
 
-  constructor(
-    private treatmentsState: TreatmentsState,
-    private mapConfig: MapConfigState,
-    private route: ActivatedRoute,
-    private router: Router
-  ) {
-    this.router.events
-      .pipe(
-        untilDestroyed(this),
-        filter((event) => event instanceof NavigationEnd) // Only react to navigation events
-      )
-      .subscribe(() => {
-        const data = getMergedRouteData(this.route.snapshot);
-        const projectAreaId = data['projectAreaId'];
-
-        if (data) {
```

Comment:
this stuff was a bit confusing, moved most stuff to loadTreatmentByRouteData and refactored treatmentsState a bit to hopefully clarify!